### PR TITLE
Fixes #8618: Unsupported key types are silently ignored in ssh key management technique

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/8618

### CODE REVIEW

File: techniques/systemSettings/remoteAccess/sshKeyDistribution/3.0/sshKeyDistribution.st [hunk 1]
```diff
@@ -52,9 +52,13 @@ bundle agent check_ssh_key_distribution
       "homedir[${sshkey_distribution_index}]"
         string  => "${userarray_${sshkey_distribution_index}[${sshkey_distribution_name[${sshkey_distribution_index}]}][5]}";
 
+      "ssh_types" string => "ssh-rsa|ssh-dss|ssh-ed25519|ssh-ecdsa";
```
Comment (amousset) [inline techniques/systemSettings/remoteAccess/sshKeyDistribution/3.0/sshKeyDistribution.st]: Could you add the list of supported key types to the parameter doc in metadata.xml?

File: techniques/systemSettings/remoteAccess/sshKeyDistribution/3.0/sshKeyDistribution.st [hunk 2]
```diff
@@ -73,11 +77,13 @@ bundle agent check_ssh_key_distribution
         string  => "${userarray_${sshkey_distribution_index}[${sshkey_distribution_name[${sshkey_distribution_index}]}][3]}";
 
   classes:
-
+      "pass1" expression => "any";
```
Comment (amousset) [inline techniques/systemSettings/remoteAccess/sshKeyDistribution/3.0/sshKeyDistribution.st]: the order of passes should be reverse

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

File: techniques/systemSettings/remoteAccess/sshKeyDistribution/3.0/sshKeyDistribution.st (hunk 1 above)
Comment (Fdall) [reply techniques/systemSettings/remoteAccess/sshKeyDistribution/3.0/sshKeyDistribution.st]: I also add ssh-dsa which may be still in use even if it is deprecated.

**Pull request merged**